### PR TITLE
Update the "Resources" section in the docs

### PR DESCRIPTION
**Pull request by mehtavishwa30**
Closes #15339 
Closes #15234 
Closes #15575

### CODE REVIEW

File: docs/resources.rst [hunk 1]
```diff
@@ -19,23 +19,28 @@ General Resources
...
+    * `Brownie <https://eth-brownie.readthedocs.io/en/stable/>`_
+        Python-based development and testing framework for smart contracts targeting the Ethereum Virtual Machine.
+
+    * `Ape <https://docs.apeworx.io/ape/stable/az>`_
+        Python-based web3 development tool for compiling, testing, and interacting with smart contracts.
+
     * `Truffle <https://trufflesuite.com/truffle/>`_
         Ethereum development framework.
+        💡 Note: Consensys announced the sunset of Truffle on September 21, 2023.
+        Current users may check out the migration path and available product support [here](https://consensys.io/blog/consensys-announces-the-sunset-of-truffle-and-ganache-and-new-hardhat?utm_source=github&utm_medium=referral&utm_campaign=2023_Sep_truffle-sunset-2023_announcement_).
```
Comment (nikola-matic) [inline docs/resources.rst]:
```suggestion
        Current users may check out the migration path and available product support `here. <https://consensys.io/blog/consensys-announces-the-sunset-of-truffle-and-ganache-and-new-hardhat>`_
```

File: docs/resources.rst [hunk 2]
```diff
@@ -19,23 +19,28 @@ General Resources
...
-    * `Hardhat <https://hardhat.org/>`_
-        Ethereum development environment with local Ethereum network, debugging features and plugin ecosystem.
-
     * `Remix <https://remix.ethereum.org/>`_
         Browser-based IDE with integrated compiler and Solidity runtime environment without server-side components.
 
+    * `Dapp <https://dapp.tools/>`_
+        Tool for building, testing and deploying smart contracts from the command-line.
+
+    * `Brownie <https://eth-brownie.readthedocs.io/en/stable/>`_
```
Comment (nikola-matic) [inline docs/resources.rst]: Brownie is no longer maintained either, an is replaced by Ape. https://eth-brownie.readthedocs.io/en/stable/

Also, why did you move the entries around - it seems like they were in alphabetical order previously?

File: docs/resources.rst [hunk 3]
```diff
@@ -140,6 +145,9 @@ Solidity Tools
 * `Universal Mutator <https://github.com/agroce/universalmutator>`_
     A tool for mutation generation, with configurable rules and support for Solidity and Vyper.
 
+* `ethdebug <https://github.com/ethdebug/format>`_
```
Comment (nikola-matic) [inline docs/resources.rst]: This one should also be moved up to maintain alphabetical order.

File: docs/resources.rst (hunk 2 above)
Comment (mehtavishwa30) [reply docs/resources.rst]: I'll double check if it indeed was the rationale but I think I moved them around to be in the order of most used based on our survey. Do you thing that's better or should I change it back to the alphabetical order?

File: docs/resources.rst (hunk 3 above)
Comment (mehtavishwa30) [reply docs/resources.rst]: Moved it up

File: docs/resources.rst (hunk 2 above)
Comment (nikola-matic) [reply docs/resources.rst]: Unless we specifically agreed to change the ordering based on survey popularity, we should just keep the alphabetical order like we do almost everywhere else - after all, the alphabet is not gonna change, whereas framework popularity most certainly will. Also, strange that we don't have Foundry here, which I'm pretty sure is no. 2, behind Hardhat (maybe even no. 1 now).

Comment (mehtavishwa30) [reply docs/resources.rst]: Ahh, you're right. Let's not make a change that would mean we'd have to update it every year according to the survey. I'll rearrange it back in the previous order and check why Foundry is not there.

EDIT: Foundry, in fact, is second on the list.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `d7acf86da2b0a8d7db347b43ff18fcea125e864b`:
```
Error: Trailing whitespace found:
docs/resources.rst:41:        Browser-based IDE with integrated compiler and Solidity runtime environment without server-side components.    

```
Please check that your changes are working as intended.

**Comment by mehtavishwa30** [comment]
Relevant

**Review by r0qs — DISMISSED** [review]
The changes look good to me now, we just need to squash the commits. I was also wondering if we could merge the changes proposed in this PR: https://github.com/ethereum/solidity/pull/15339, or possibly even close it. I haven’t looked too closely at the tools being proposed - although the static analyzer seems nice - but since we're working on the same docs and there’s some overlap in the changes, this might be a good time to make a decision on that.

File: docs/resources.rst [hunk 4]
```diff
@@ -5,22 +5,28 @@ Resources
...
-* `Solidity Source Code on GitHub <https://github.com/ethereum/solidity/>`_
-* `Solidity Language Users Chat <https://matrix.to/#/#ethereum_solidity:gitter.im>`_
-* `Solidity Compiler Developers Chat <https://matrix.to/#/#ethereum_solidity-dev:gitter.im>`_
-* `Awesome Solidity <https://github.com/bkrem/awesome-solidity>`_
+* `Solidity website <https://soliditylang.org/>`_
+* `Solidity changelog <https://github.com/ethereum/solidity/blob/develop/Changelog.md>`_
+* `Solidity codebase on GitHub <https://github.com/ethereum/solidity/>`_
+* `Solidity language users chat <https://matrix.to/#/#ethereum_solidity:gitter.im>`_
+* `Solidity compiler developers chat <https://matrix.to/#/#ethereum_solidity-dev:gitter.im>`_
+* `awesome-solidity <https://github.com/bkrem/awesome-solidity>`_
```
Comment (r0qs) [inline docs/resources.rst:15]: This still not in alphabetical order, but I guess the idea was to keep it as it was before, right?

**Comment by mehtavishwa30** [comment]
> The changes look good to me now, we just need to squash the commits. I was also wondering if we could merge the changes proposed in this PR: #15339, or possibly even close it. I haven’t looked too closely at the tools being proposed - although the static analyzer seems nice - but since we're working on the same docs and there’s some overlap in the changes, this might be a good time to make a decision on that.

@r0qs Thanks for the review. I wanted to get to this today. Just caught up in some other issues. The idea is to consolidate the changes proposed in the external contribution, so I will ask for a final review once I've included those. :)

**Comment by r0qs** [comment]
> > The changes look good to me now, we just need to squash the commits. I was also wondering if we could merge the changes proposed in this PR: #15339, or possibly even close it. I haven’t looked too closely at the tools being proposed - although the static analyzer seems nice - but since we're working on the same docs and there’s some overlap in the changes, this might be a good time to make a decision on that.
> 
> @r0qs Thanks for the review. I wanted to get to this today. Just caught up in some other issues. The idea is to consolidate the changes proposed in the external contribution, so I will ask for a final review once I've included those. :)

No problem. But yeah, that was just a suggestion though, because we have so many documentation PRs that have small changes, that maybe we could combine some of them. For example, this may also be relevant to add here or just close it: https://github.com/ethereum/solidity/pull/15234

File: docs/resources.rst (hunk 4 above)
Comment (mehtavishwa30) [reply docs/resources.rst:15]: That's correct. Just made minor naming changes in the title that I thought were more appropriate. But otherwise, I'm not strongly set on the order.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `89489df1308512ba3b3da3413ea94d5691e22a9e`:
```
Error: Trailing whitespace found:
docs/resources.rst:74:* Visual Studio Code (VS Code)    

```
Please check that your changes are working as intended.

File: docs/resources.rst [hunk 5]
```diff
@@ -66,10 +71,10 @@ Editor Integrations
     * `Vim Syntastic <https://github.com/vim-syntastic/syntastic>`_
         Plugin for the Vim editor providing compile checking.
 
-* Visual Studio Code (VS Code)
-
+* Visual Studio Code (VS Code)
```
Comment (nikola-matic) [inline docs/resources.rst]:
```suggestion
* Visual Studio Code (VS Code)
```

File: docs/resources.rst [hunk 6]
```diff
@@ -5,58 +5,63 @@ Resources
...
-* `Solidity Portal <https://soliditylang.org/>`_
-* `Solidity Changelog <https://github.com/ethereum/solidity/blob/develop/Changelog.md>`_
-* `Solidity Source Code on GitHub <https://github.com/ethereum/solidity/>`_
-* `Solidity Language Users Chat <https://matrix.to/#/#ethereum_solidity:gitter.im>`_
-* `Solidity Compiler Developers Chat <https://matrix.to/#/#ethereum_solidity-dev:gitter.im>`_
-* `Awesome Solidity <https://github.com/bkrem/awesome-solidity>`_
+* `Solidity website <https://soliditylang.org/>`_
+* `Solidity changelog <https://github.com/ethereum/solidity/blob/develop/Changelog.md>`_
+* `Solidity codebase on GitHub <https://github.com/ethereum/solidity/>`_
+* `Solidity language users chat <https://matrix.to/#/#ethereum_solidity:gitter.im>`_
```
Comment (nikola-matic) [inline docs/resources.rst:13]: Why did you move this to lower caps? The previous style is more consistent with the rest of our docs.

Comment (mehtavishwa30) [reply docs/resources.rst:13]: it just looked a little weird, IMO, in that earlier format. But I don't think it's too big of a deal. LMK if you think I should change it back or if it's okay as is now.

File: docs/resources.rst [hunk 7]
```diff
@@ -5,58 +5,63 @@ Resources
...
+        💡 Note: Consensys announced the sunset of Truffle on September 21, 2023.
+        Current users may check out the migration path and available product support `here.
+        <https://consensys.io/blog/consensys-announces-the-sunset-of-truffle-and-ganache-and-new-hardhat>`_
 
 Editor Integrations
 ===================
 
 * Emacs
-
     * `Emacs Solidity <https://github.com/ethereum/emacs-solidity/>`_
```
Comment (cameel) [inline docs/resources.rst]: Please make sure this renders correctly before merging. In RST there must be an empty line between a bullet list and the surrounding text. If it's not there, the list gets mangled into a single paragraph as if it was a continuation of the previous sentence. Not sure if it is needed in this particular case, but if it was here already, it might have been for this very reason.

**Review by cameel — COMMENTED** [review]
Note that CI is failing (it's the spellcheck job - you have a spelling mistake somewhere in there).

Other than that it looks good. I haven't reviewed this in detail, but it's been sitting there for a long time and it would be nice to finally get this merged.

**Comment by cameel** [comment]
BTW, you could list all the other redundant PRs in your description and github will automatically close them when we merge this one. That would make it easier to keep track of them, because we have quite a few now.

To do this write something like `Closes #xyz`. Just note that it has to be in the description at the top (not in a random comment) and you have to repeat `Closes` for each one (so no shortcuts like `Closes #xyz, #abc, #def`).

File: docs/resources.rst (hunk 7 above)
Comment (mehtavishwa30) [reply docs/resources.rst]: I just fixed this, I think it should be okay now. Let me know if this is good to merge.

**Review by cameel — APPROVED** [review]
Looks good to me, but needs some commit squashing. I guess we could just squash-merge?
